systemtest/src/main/java/io/strimzi/systemtest/performance/report/parser/PerformanceMetricsParser.java

```diff
@@ -0,0 +1,392 @@
+                    yamlSectionStarted = true; // Start collecting YAML content
+                } else if (!yamlSectionStarted) {
+                    // Process simple key-value pairs
+                    String[] parts = line.split(": ", 2);
+                    if (parts.length == 2) {
+                        experimentMetrics.addSimpleMetric(parts[0].trim(), parts[1].trim());
+                    }
+                } else {
+                    // Collect YAML content
+                    yamlBuilder.append(line).append("\n");
+                }
+            }
+        }
+
+        if (yamlBuilder.length() > 0) {
+            ObjectMapper yamlMapper = new YAMLMapper();
+            KafkaSpec kafkaSpec = yamlMapper.readValue(yamlBuilder.toString(), KafkaSpec.class);
+            experimentMetrics.setKafkaSpec(kafkaSpec);
+        }
+    }
+
+    // Helper method to parse the list of doubles from a String and find the max value
+    private static double getMaxValueFromList(String listAsString) {
+        // Assuming the list is formatted as "[value1, value2, ...]"
+        String[] items = listAsString.substring(1, listAsString.length() - 1).split(", ");
+        return Arrays.stream(items)
+                .mapToDouble(Double::parseDouble)
+                .max()
+                .orElse(0); // Default to 0 if list is empty or parsing fails
+    }
+
+    public static void showValuesOfExperiments() {
+        // Populate data for each experiment
+        USE_CASE_EXPERIMENTS.forEach((useCaseName, experimentsList) -> {
+            System.out.println("Use Case: " + useCaseName);
+
+            String[] headers = getHeadersForUseCase(useCaseName);
+            List<String[]> allRows = new ArrayList<>();
+
+            allRows.add(headers);
+
+            // Determine max width for each column
+            int[] columnWidths = new int[headers.length];
+            for (String[] row : allRows) {
+                for (int i = 0; i < row.length; i++) {
+                    columnWidths[i] = Math.max(columnWidths[i], row[i].length());
+                }
+            }
+
+            printSeparator(columnWidths);
+
+            int experimentCounter = 1;
+
+            for (ExperimentMetrics experimentMetrics : experimentsList) {
+                Map<String, String> simpleMetrics = experimentMetrics.getSimpleMetrics();
+                // Assume methods to extract and format metrics correctly are implemented
+                String[] rowData = extractAndFormatRowData(experimentCounter, simpleMetrics, experimentMetrics);
+                allRows.add(rowData);
+                experimentCounter++;
+            }
+
+            // Print table with dynamically adjusted widths
+            allRows.forEach(row -> printRow(row, columnWidths));
+            printSeparator(columnWidths);
+        });
+    }
+
+    private static String[] extractAndFormatRowData(int experimentNumber, Map<String, String> simpleMetrics, ExperimentMetrics experimentMetrics) {
```

Comment:
As you mentioned, if methods like these are related just to TO, you should have separate class for it, or add possibility to specify the metrics etc. Because otherwise, you should rename the class - which would contradict with the general methods you have here.

Reply:
+1